### ISSUE TITLE
Sorry to tell everyone but I currently won't maintain this repo anymore

### ISSUE THREAD

**Issue by amtlib-dot-dll** [flathub/com.visualstudio.code.oss#46]
@nedrichards @TingPing @barthalion @alexlarsson @ebassi @jurf 

I have recently graduated from campus and found a job, and I don't have much spare time as before. Missing my school days. 

This repo consists of a Python script generating a JSON recipe. Though not fully documented, the symbol names should be pretty self-explanatory. 

Feel free to ask me about any further questions about it. I will check my GitHub inbox irregularly. 

Wishing volunteers for our project and our dream for free softwares!

**Comment by barthalion** [comment]
I don't think there will be many people happy to pick up maintenance so it should be marked EOL and probably set `end-of-life-rebase` to regular VS Code.

**Comment by G-Ray** [comment]
Thank you for all the time you have spent on this project @amtlib-dot-dll !

Even if this project (and vscodium) is a step towards a more free software than official vscode binary, as long as these projects still rely on the microsoft marketplace it can't be really "libre".

It would be awesome if a team would create a true libre project based on vscode, with a reimplementation of the marketplace's backend which would only contains libre extensions.

**Comment by HarryMichal** [comment]
Isn't this a great opportunity to become official flatpak of VSCodium[1]? I can try to set it up.

[1] https://vscodium.com/

**Comment by G-Ray** [comment]
Exactly @HarryMichal, you can start by forking the official vscode from flathub repo (https://github.com/flathub/com.visualstudio.code). I started to work on it but did not have much time for now

**Comment by amtlib-dot-dll** [comment]
Good deal @G-Ray and @HarryMichal! By the way I'd like to remind you that the building process requires network isolation, and therefore a lot of work of the current recipe is to pre-calculate the dependencies. @barthalion Would there be any exception for that?

**Comment by gasinvein** [comment]
Hello everyone. I'd like to revive VS Code OSS on Flathub, but the build system of this flatpak feels way too complicated, at least for me. So I've made it almost from scratch, without any custom manifest generation scripts involved; instead, the `flatpak-node-generator.py` script does all the hard work. If anybody interested, have a look at the [new manifest](https://github.com/gasinvein/flatpak-apps-staging/tree/com.visualstudio.code-oss/master).

**Comment by TingPing** [comment]
@gasinvein I've given you access, have at it :)

**Comment by gasinvein** [comment]
@TingPing Honestly, I'd rather like to submit a new app with different ID (`.code-oss` instead of `.code.oss`). There are multiple reasons to do so:
- `code-oss` is the actual base name that MS uses for OSS builds (e.g. it's icons and .desktop files)
- This makes sure we don't clash with user data from existing installations (which might be incompatible)
- It is an entirely different flatpak build, and I don't want to just replace @amtlib-dot-dll's work with mine

**Comment by TingPing** [comment]
> It is an entirely different flatpak build, and I don't want to just replace @amtlib-dot-dll's work with mine

This is irrelevant. Its just a package.

If the app-id is different upstream, and I mean it actually is fully `com.visualstudio.code-oss`, then thats fair. It should still be rebased automatically to the new app.

**Comment by Croydon** [comment]
Would there be any differences to https://flathub.org/apps/details/com.vscodium.codium ?

**Comment by gasinvein** [comment]
@Croydon AFAIK vscodium is just a binary build of vscode-oss. So, the vscodium flatpak repacks those binary builds, while this flatpak (and so does mine) builds it from source. That said, you should ask vscodium maintainers how their builds differ from VS Code builds from original source.

**Comment by gasinvein** [comment]
@TingPing I didn't notice MS using reverse-dns notation in VS Code OSS, but `code-oss` is used everywhere in desktop data.
>It should still be rebased automatically to the new app.

Yes, it's what I intended. Submit a new app with new ID, then mark this one as EOL and replace it with the new one via `end-of-life-rebase`.

**Comment by TingPing** [comment]
> @TingPing I didn't notice MS using reverse-dns notation in VS Code OSS, but code-oss is used everywhere in desktop data.

In my opinion the app-id change is pointless then. Its not like we are matching upstream everything will still be renamed.

**Comment by gasinvein** [comment]
@TingPing It's still the easiest way to deal with existing user data, which, as I noted, might not be compatible. By chaining ID we don't touch existing installations at all and thus no chance to break it.

**Comment by TingPing** [comment]
Why would compatibility change?

**Comment by gasinvein** [comment]
Because it's built in a different way, probably contains slightly different npm packages, uses wrapper from the proprietary vscode flatpak, and is several major versions newer.

**Comment by TingPing** [comment]
"It updated" is going to happen no matter what, I'm not sure I understand the fear unless you experienced something broke compat.

**Comment by gasinvein** [comment]
@TingPing There are many internal bits and pieces that are different. I really don't want to dig into it to discover waht's different in resulting app and how exactly it can affect behavior. _Maybe_ it's actually safe an nothing will break - but if not, who's going to fix it and how? Starting with clean data/config directories, on the other hand, is always-safe option.

**Comment by TingPing** [comment]
You are putting in the effort do whatever I suppose. It just seems like making a problem out of nothing to me.

**Comment by gasinvein** [comment]
No, I'm trying to avoid problems. If creating yet another repo under flathub org is a problem - well then, will submit the new build here. Arguing with you seems pointless anyway.

**Comment by TingPing** [comment]
Sorry if I came across as arguing, I was just confused about it.

**Comment by gasinvein** [comment]
Anyone interested - please give #56 a try and tell if everything works as expected.

**Comment by mzabaluev** [comment]
Chiming in to the discussion about the app id: it might be problematic in the long run to use the name prefix that is technically controlled by a different entity (it's borderline OK for `com.visualstudio.code` as long as it repacks the official binaries plus the supporting dependencies).
I'm proposing a convention for cases like this in https://github.com/flathub/flathub/issues/1595.
